File: data/multilang_comparison/program_cpp_vector/src/func4.cpp
#include "program_cpp_vector.hpp" 
Array* func4(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array110;
   if (pCounter > 0) {
      array110 = vars->data.at(--pCounter);
      array110->refC++;
      DEBUG_COPY(array110->id);
   } else {
      array110 = new Array();
      array110->size = 209;
      array110->refC = 1;
      array110->id = 110;
      array110->data = std::vector<unsigned int>(array110->size, 0);
      DEBUG_NEW(array110->id);
   }
   unsigned int loop33 = 0;
   unsigned int loopLimit33 = (50)/3 + 1;
   for(; loop33 < loopLimit33; loop33++) {
      if(PATH0 & 1) {
         Array* array111;
         if (pCounter > 0) {
            array111 = vars->data.at(--pCounter);
            array111->refC++;
            DEBUG_COPY(array111->id);
         } else {
            array111 = new Array();
            array111->size = 581;
            array111->refC = 1;
            array111->id = 111;
            array111->data = std::vector<unsigned int>(array111->size, 0);
            DEBUG_NEW(array111->id);
         }
         unsigned int loop34 = 0;
         unsigned int loopLimit34 = (50)/4 + 1;
         for(; loop34 < loopLimit34; loop34++) {
            for (int i = 0; i < array111->size; i++) {
                array111->data.at(i)--;
            }
            for (int i = 0; i < array110->size; i++) {
                if (array110->data.at(i) == 32) {
                    return array110;
                }
            }
         }
         Array_param params0;
         params0.size = 2;
         std::vector<Array*> data_params0(2);
         params0.data = data_params0;
         params0.data.at(0) = array110;
         params0.data.at(1) = array111;
         Array* array112 = func24(&params0, get_path(), loopsFactor);
         DEBUG_RETURN(array112->id);
         if(params0.size > 0) {
         	 params0.data.clear();
         }
         array112->refC--;
         if(array112->refC == 0) {
            array112->data.clear();
            delete array112;
            DEBUG_FREE(array112->id);
         }
         array111->refC--;
         if(array111->refC == 0) {
            array111->data.clear();
            delete array111;
            DEBUG_FREE(array111->id);
         }
      }
      else {
         Array_param params0;
         params0.size = 1;
         std::vector<Array*> data_params0(1);
         params0.data = data_params0;
         params0.data.at(0) = array110;
         Array* array113 = func16(&params0, get_path(), loopsFactor);
         DEBUG_RETURN(array113->id);
         if(params0.size > 0) {
         	 params0.data.clear();
         }
         array113->refC--;
         if(array113->refC == 0) {
            array113->data.clear();
            delete array113;
            DEBUG_FREE(array113->id);
         }
      }
      Array* array114;
      if (pCounter > 0) {
         array114 = vars->data.at(--pCounter);
         array114->refC++;
         DEBUG_COPY(array114->id);
      } else {
         array114 = new Array();
         array114->size = 769;
         array114->refC = 1;
         array114->id = 114;
         array114->data = std::vector<unsigned int>(array114->size, 0);
         DEBUG_NEW(array114->id);
      }
      Array_param params0;
      params0.size = 2;
      std::vector<Array*> data_params0(2);
      params0.data = data_params0;
      params0.data.at(0) = array110;
      params0.data.at(1) = array114;
      Array* array115 = func17(&params0, loopsFactor);
      DEBUG_RETURN(array115->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      if(PATH0 & 2) {
         Array* array116;
         if (pCounter > 0) {
            array116 = vars->data.at(--pCounter);
            array116->refC++;
            DEBUG_COPY(array116->id);
         } else {
            array116 = new Array();
            array116->size = 850;
            array116->refC = 1;
            array116->id = 116;
            array116->data = std::vector<unsigned int>(array116->size, 0);
            DEBUG_NEW(array116->id);
         }
         unsigned int loop35 = 0;
         unsigned int loopLimit35 = (50)/4 + 1;
         for(; loop35 < loopLimit35; loop35++) {
            for (int i = 0; i < array116->size; i++) {
                array116->data.at(i)--;
            }
            for (int i = 0; i < array110->size; i++) {
                if (array110->data.at(i) == 42) {
                    return array110;
                }
            }
         }
         Array_param params1;
         params1.size = 4;
         std::vector<Array*> data_params1(4);
         params1.data = data_params1;
         params1.data.at(0) = array110;
         params1.data.at(1) = array114;
         params1.data.at(2) = array115;
         params1.data.at(3) = array116;
         Array* array117 = func28(&params1, get_path(), loopsFactor);
         DEBUG_RETURN(array117->id);
         if(params1.size > 0) {
         	 params1.data.clear();
         }
         array117->refC--;
         if(array117->refC == 0) {
            array117->data.clear();
            delete array117;
            DEBUG_FREE(array117->id);
         }
         array116->refC--;
         if(array116->refC == 0) {
            array116->data.clear();
            delete array116;
            DEBUG_FREE(array116->id);
         }
      }
      else {
         Array_param params1;
         params1.size = 3;
         std::vector<Array*> data_params1(3);
         params1.data = data_params1;
         params1.data.at(0) = array110;
         params1.data.at(1) = array114;
         params1.data.at(2) = array115;
         Array* array118 = func22(&params1, get_path(), loopsFactor);
         DEBUG_RETURN(array118->id);
         if(params1.size > 0) {
         	 params1.data.clear();
         }
         array118->refC--;
         if(array118->refC == 0) {
            array118->data.clear();
            delete array118;
            DEBUG_FREE(array118->id);
         }
      }
      for (int i = 0; i < array115->size; i++) {
          if (array115->data.at(i) == 67) {
              return array115;
          }
      }
      array115->refC--;
      if(array115->refC == 0) {
         array115->data.clear();
         delete array115;
         DEBUG_FREE(array115->id);
      }
      array114->refC--;
      if(array114->refC == 0) {
         array114->data.clear();
         delete array114;
         DEBUG_FREE(array114->id);
      }
   }
   for (int i = 0; i < array110->size; i++) {
       array110->data.at(i)--;
   }
   for (int i = 0; i < array110->size; i++) {
       if (array110->data.at(i) == 61) {
           return array110;
       }
   }
   return array110;
}

File: data/multilang_comparison/program_go/src/func4.go
package main
import "math/rand"
var _ = rand.Intn
func Func4(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array110 *Array
   if pCounter > 0 {
       array110 = vars.Data[pCounter-1]
       pCounter--
       array110.RefC++
   } else {
       array110 = &Array{
           Size: 209,
           RefC: 1,
           Id: 110,
           Data: make([]uint32, 209),
       }
   }
   var loop33 int = 0;
   var loopLimit33 int = (50)/3 + 1;
   for ; loop33 < loopLimit33; loop33++ {
      if (PATH0 & 1) != 0 {
         var array111 *Array
         if pCounter > 0 {
             array111 = vars.Data[pCounter-1]
             pCounter--
             array111.RefC++
         } else {
             array111 = &Array{
                 Size: 581,
                 RefC: 1,
                 Id: 111,
                 Data: make([]uint32, 581),
             }
         }
         var loop34 int = 0;
         var loopLimit34 int = (50)/4 + 1;
         for ; loop34 < loopLimit34; loop34++ {
            for i := 0; i < array111.Size; i++ {
                array111.Data[i]--
            }
            for i := 0; i < array110.Size; i++ {
                if array110.Data[i] == 32 {
                    return array110
                }
            }
         }
         var params0 ArrayParam
         params0.Size = 2
         params0.Data = make([]*Array, params0.Size)
         params0.Data[0] = array110
         params0.Data[1] = array111
         var array112 *Array = Func24(&params0, getPath(), loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array112.RefC--
         if array112.RefC == 0 {
             array112.Data = nil
             array112 = nil
         }
         array111.RefC--
         if array111.RefC == 0 {
             array111.Data = nil
             array111 = nil
         }
      
      } else {
         var params0 ArrayParam
         params0.Size = 1
         params0.Data = make([]*Array, params0.Size)
         params0.Data[0] = array110
         var array113 *Array = Func16(&params0, getPath(), loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array113.RefC--
         if array113.RefC == 0 {
             array113.Data = nil
             array113 = nil
         }
      }
      var array114 *Array
      if pCounter > 0 {
          array114 = vars.Data[pCounter-1]
          pCounter--
          array114.RefC++
      } else {
          array114 = &Array{
              Size: 769,
              RefC: 1,
              Id: 114,
              Data: make([]uint32, 769),
          }
      }
      var params0 ArrayParam
      params0.Size = 2
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array110
      params0.Data[1] = array114
      var array115 *Array = Func17(&params0, loopsFactor);
      params0.Data = nil
      params0.Size = 0
      if (PATH0 & 2) != 0 {
         var array116 *Array
         if pCounter > 0 {
             array116 = vars.Data[pCounter-1]
             pCounter--
             array116.RefC++
         } else {
             array116 = &Array{
                 Size: 850,
                 RefC: 1,
                 Id: 116,
                 Data: make([]uint32, 850),
             }
         }
         var loop35 int = 0;
         var loopLimit35 int = (50)/4 + 1;
         for ; loop35 < loopLimit35; loop35++ {
            for i := 0; i < array116.Size; i++ {
                array116.Data[i]--
            }
            for i := 0; i < array110.Size; i++ {
                if array110.Data[i] == 42 {
                    return array110
                }
            }
         }
         var params1 ArrayParam
         params1.Size = 4
         params1.Data = make([]*Array, params1.Size)
         params1.Data[0] = array110
         params1.Data[1] = array114
         params1.Data[2] = array115
         params1.Data[3] = array116
         var array117 *Array = Func28(&params1, getPath(), loopsFactor);
         params1.Data = nil
         params1.Size = 0
         array117.RefC--
         if array117.RefC == 0 {
             array117.Data = nil
             array117 = nil
         }
         array116.RefC--
         if array116.RefC == 0 {
             array116.Data = nil
             array116 = nil
         }
      
      } else {
         var params1 ArrayParam
         params1.Size = 3
         params1.Data = make([]*Array, params1.Size)
         params1.Data[0] = array110
         params1.Data[1] = array114
         params1.Data[2] = array115
         var array118 *Array = Func22(&params1, getPath(), loopsFactor);
         params1.Data = nil
         params1.Size = 0
         array118.RefC--
         if array118.RefC == 0 {
             array118.Data = nil
             array118 = nil
         }
      }
      for i := 0; i < array115.Size; i++ {
          if array115.Data[i] == 67 {
              return array115
          }
      }
      array115.RefC--
      if array115.RefC == 0 {
          array115.Data = nil
          array115 = nil
      }
      array114.RefC--
      if array114.RefC == 0 {
          array114.Data = nil
          array114 = nil
      }
   }
   for i := 0; i < array110.Size; i++ {
       array110.Data[i]--
   }
   for i := 0; i < array110.Size; i++ {
       if array110.Data[i] == 61 {
           return array110
       }
   }
   return array110;
}

